backend/src/Controller/RegistrationController.php

```
@@ -94,7 +104,47 @@ public function register(
+        // reach this account, and one that only exists after its owner next
+        // visits the Sharing page cannot be shared with until then.
         $entityManager->persist($user);
+
+        try {
+            $entityManager->flush();
+        } catch (UniqueConstraintViolationException) {
+            // Two registrations picked the same username between the check and
+            // this write. The index is the authority and has just said so;
+            // Doctrine closes the manager on it, so the useful answer is a
+            // fresh suggestion rather than a retry this request cannot make.
+            return new JsonResponse([
```

PHPStan (2.2.7)

[error] Method App\Controller\RegistrationController::register() should return Symfony\Component\HttpFoundation\Response but returns Symfony\Component\HttpFoundation\JsonResponse.


(return.type)